### PR TITLE
Dockerise entire app

### PR DESCRIPTION
**Pull request by dct0**
## Change Summary
**[Briefly summarise the changes that you made. Just high-level stuff]**

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [X] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -10,31 +11,40 @@ services:
       timeout: 3s
       retries: 5
     env_file:
-      - ./server/.env
+      - path: ./server.env
+        required: false
+      - ./server/.env.example
```
Comment (frinzekt) [inline docker-compose.yml:16]: Isn't this in order? Top to bottom?

So the example will always overwrite the previous...

This is a bit wrong 😅

File: .devcontainer/install-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+cd server && POETRY_VIRTUALENVS_CREATE=false poetry install # give editor intellisense
```
Comment (frinzekt) [inline .devcontainer/install-dependencies.sh:3]: Add docker compose build...

So when they run docker compose up, it's faster

File: docker-compose.yml (hunk 1 above)
Comment (dct0) [reply docker-compose.yml:16]: > I would kinda prefer that you copy the .Env file like I did in my pr

agreed

**Review by frinzekt — APPROVED** [review]

File: .devcontainer/install-dependencies.sh (hunk 2 above)
Comment (frinzekt) [reply .devcontainer/install-dependencies.sh:3]: Put the cd server and the other command in the parentheses, so it doesn't affect your current working directory

File: .devcontainer/setup.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+fi
+if [ ! -f ./server/.env ]; then
+    cp ./server/.env.example ./server/.env
+fi
+
+# Give python editor intellisense
+(cd server && POETRY_VIRTUALENVS_CREATE=false poetry install)
+
+# Build containers so less waiting
+docker compose pull
```
Comment (frinzekt) [inline .devcontainer/setup.sh:20]: Pull is different from build...

Pull will only pull images works if you don't have dockerfiles, but doesn't not do build.

Probably do both!

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -12,13 +12,13 @@
...
 
   // Use 'forwardPorts' to make a list of ports inside the container available locally.
-  "forwardPorts": [3000, 8000, 8080]
+  "forwardPorts": [3000, 8000, 8080],
 
   // Use 'postCreateCommand' to run commands after the container is created.
   // Note: Useful for when they open it and want everything to just start
-  // "postCreateCommand": "docker compose up -d"
+  "postCreateCommand": "bash ./.devcontainer/setup.sh"
 }
```
Comment (frinzekt) [inline .devcontainer/devcontainer.json:24]: If you don't want running bash, you can do chmod 755 on the shell script, so it becomes an executable

**Pull request merged**